src/v/iceberg/manifest_list_avro.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_list_avro.h"
+
+#include "base/units.h"
+#include "bytes/iobuf.h"
+#include "iceberg/avro_utils.h"
+#include "iceberg/manifest_file.avrogen.h"
+#include "strings/string_switch.h"
+
+#include <avro/DataFile.hh>
+
+#include <stdexcept>
+#include <vector>
+
+namespace iceberg {
+
+namespace {
+
+constexpr int32_t content_type_to_int(manifest_file_content t) {
+    switch (t) {
+    case manifest_file_content::data:
+        return 1;
+    case manifest_file_content::deletes:
+        return 2;
```

Comment:
don't the enum types already have assigned integral values?

Reply:
I can assign them. They won't start at 0 though since these need to match the Iceberg spec

Reply:
Ah I take it back about starting at 0. Not sure why I had these start at 1, but the spec starts at 0. Updating!